stack-graphs/src/arena.rs

```diff
@@ -161,64 +161,99 @@ unsafe impl<T> Sync for Handle<T> {}
 /// the arena.  All of the instances managed by this arena will be dropped as a single operation
 /// when the arena itself is dropped.
 pub struct Arena<T> {
-    items: Vec<MaybeUninit<T>>,
+    chunks: Vec<Vec<MaybeUninit<T>>>,
```

Comment:
Would there be drawbacks to using `Vec<Box<[T; CHUNK_SIZE]>>`? It would make it more obvious that the inner elements are fixed sized, and accidentally extending them would be a hard error.

Reply:
Makes sense!

Reply:
Thinking through this more, though, I think you're on to something.  We don't actually need the "used size" of _every_ chunk, just the last one!  All of the other chunks will be completely full.  So we could just store the "used size" of the arena as a whole.  That might make some of the calculations simpler!

Reply:
This was a good idea, thanks @hendrikvanantwerpen!  We're using `Box<[]>` instead of `Vec` now for the inner chunks.